### PR TITLE
[WFLY-14926] Upgrade archetypes to WildFly 24

### PR DESCRIPTION
**Pull request by WolfgangHG**
Issue: https://issues.redhat.com/browse/WFLY-14926

### CODE REVIEW

**Comment by emmartins** [comment]
@WolfgangHG it's ok to not have the jira assigned to you, I can do such change once you submit the PR.

Btw, here are the general important guidelines for the PR:
* The commit should have the JIRA key (as you did)
* The PR description should have a line" Issue: URL_TO_THE_JIRA"

**Comment by emmartins** [comment]
@WolfgangHG Did you register a JIRA user, I can't seem to find you to assign

File: wildfly-jakartaee-webapp-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
         <version.wildfly.maven.plugin>2.0.2.Final</version.wildfly.maven.plugin>
 
         <!-- Define the version of the JBoss BOMs we want to import to specify tested stacks. -->
-        <version.jboss.bom>23.0.0.Final</version.jboss.bom>
+        <version.jboss.bom>24.0.0.Final</version.jboss.bom>
```
Comment (emmartins) [inline wildfly-jakartaee-webapp-archetype/src/main/resources/archetype-resources/pom.xml:40]: BOMs should be 24.0.0.SP1, the 24.0.0.Final ones had issues

File: wildfly-jakartaee-ear-archetype/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -43,7 +43,7 @@
         <version.wildfly.maven.plugin>2.0.2.Final</version.wildfly.maven.plugin>
 
         <!-- Define the version of the JBoss BOMs we want to import to specify tested stacks. -->
-        <version.jboss.bom>23.0.0.Final</version.jboss.bom>
+        <version.jboss.bom>24.0.0.Final</version.jboss.bom>
```
Comment (emmartins) [inline wildfly-jakartaee-ear-archetype/src/main/resources/archetype-resources/pom.xml:46]: BOMs should be 24.0.0.SP1, the 24.0.0.Final ones had issues

**Comment by WolfgangHG** [comment]
Updated the pull request.

About JIRA: maybe my account does not have enough privileges - I also cannot assign the issue to anybody else.
Actually, I don't need to assign issues to myself, but the workflow should probably be to assign it to you to do the finale release. 
What to do about this?

**Pull request merged**